### PR TITLE
feat(python): add "pyxlsb" engine support to `read_excel` (for excel binary workbook files)

### LINKED ISSUE #11181: Read XLSX and XLSB files natively
### Description

Hello!

This project is absolutely amazing and I really appreciate when I get notifications about updates :)
So, I created a small python module to read XLSX and XLSB initially, in pure python, into PyArrow arrays, and then into dataframe, Polars or Pandas.
This is just a module concept, but it already works on my machine. It is very fast and consumes very little RAM
I started working on this when Polars didn't have built-in Excel support - only via xlsx2csv.
Firstly, I've also worked with openpyxl and it just works, but I've found that it's very slow for me and sometimes very inefficient.
I would like to participate in your project by investing a little with my code or idea.
The module is codenamed RXLS (Read XLS[X|B]), and although I have left almost no comments there, I am ready to answer any questions. There may be some issues, and I will resolve them all...
I also worked with Rust at one time, but that was quite a long time ago.

https://github.com/SaelKimberly/rxls

Thank you very much, I really appreciate your work 😃

### LINKED ISSUE #11184: read_excel schema_overrides missing
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
from openpyxl import load_workbook
from pathlib import Path

xl_file=Path("my_xl.xlsx")
wb=load_workbook(xl_file)
sheets=wb.sheetnames
wb.close()

df=pl.read_excel(xl_file, 
sheet_name = sheets[1], 
schema_overrides = {'Date Announced': pl.Date, 'Date Withdrawn': pl.Date,                                                     
'Rank Date': pl.Date}
)

```

### Log output

_No response_

### Issue description

Calling the read_excel function and trying to overwrite some data types using the schema_overrides parameter but am getting an error saying that parameter is unexpected.  Documentation shows it exists.  Was this removed?  Am I missing something?  I'm using 0.19.2, FYI

https://pola-rs.github.io/polars/py-polars/html/reference/api/polars.read_excel.html

### Expected behavior

expected behavior would include the schema to change the data type for the specified columns

### Installed versions

<details>

```
Polars:              0.19.2
Index type:          UInt32
Platform:            Windows-10-10.0.19044-SP0
Python:              3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_sqlite:  0.6.0
cloudpickle:         2.2.1
connectorx:          0.3.2a7
deltalake:           0.10.1
fsspec:              2023.5.0
matplotlib:          3.7.2
numpy:               1.24.4
pandas:              2.1.0
pyarrow:             12.0.1
pydantic:            2.1.1
sqlalchemy:          2.0.19
xlsx2csv:            0.8.1
xlsxwriter:          3.1.2
```

</details>

### PR DESCRIPTION
**Pull request by alexander-beedie**
Closes #11181 (and also closes #11184 by adding a `.. versionadded` tag to the docs).

* Adds support for the `pyxlsb` engine so that we can also read Excel Binary Workbook files (those with an ".xlsb" extension, which are not compatible with any of the existing ".xlsx" engines). Note that this engine does _not_ currently autodetect datetime/date columns (it reads them in as Excel's native offset-Julian float), and therefore _requires_ the use of `schema_overrides` to load them correctly (I have added a note in the docstring about this).

* Also: improves Date parsing for OpenOffice files via `read_ods`).

* Also: slightly updates `show_versions()` docs/example with the latest libs.

---

Support for spreadsheet data is improving nicely; with this update `read_excel` can now read all of the major Excel formats (".xlsx", ".xlsm", "xlsb"), and we can handle the OpenOffice ".ods" format via `read_ods`.

## Example
Reading ".xlsb" files:
```python
pl.read_excel(
    source = "~/test.xlsb",
    sheet_name="misc_data",
    schema_overrides={"dtm":pl.Datetime, "dt":pl.Date},
)
```
**Before**
```
xlsx2csv.XlsxException: Sheet 'misc_data' not found
```
**After**
```
shape: (2, 3)
┌─────────────────────┬────────────┬──────┐
│ dtm                 ┆ dt         ┆ val  │
│ ---                 ┆ ---        ┆ ---  │
│ datetime[μs]        ┆ date       ┆ f64  │
╞═════════════════════╪════════════╪══════╡
│ 1999-12-31 10:30:45 ┆ 2024-01-01 ┆ 1.5  │
│ 2010-10-11 12:13:14 ┆ 2018-08-07 ┆ -0.5 │
└─────────────────────┴────────────┴──────┘
```

---

FYI: @SaelKimberly has been experimenting with writing a potentially superior ".xlsb" reading engine; if & when this is ready (and well tested and available from `pypi` ;) we can look at including that as the default `.xlsb` engine instead - looking forward to it 👍

### CODE REVIEW

File: py-polars/tests/unit/io/test_spreadsheet.py [hunk 1]
```diff
@@ -223,10 +243,26 @@ def test_schema_overrides_11161(excel_file_path: Path) -> None:
...
+            schema_overrides={"dtm": pl.Datetime("ns"), "dt": pl.Date},
+        )
+        assert df4.to_dict(False) == {
+            "dtm": [
+                datetime(1999, 12, 31, 10, 30, 45),
+                datetime(2010, 10, 11, 12, 13, 14),
+            ],
+            "dt": [date(2024, 1, 1), date(2018, 8, 7)],
+            "val": [1.5, -0.5],
+        }
```
Comment (stinodego) [inline py-polars/tests/unit/io/test_spreadsheet.py:260]: Can we agree to write test assertions using our `assert_frame_equal` util? At least when working with datetimes/durations with non-standard precision, to make sure dtypes are correct.

File: py-polars/polars/utils/show_versions.py [hunk 2]
```diff
@@ -58,18 +59,20 @@ def show_versions() -> None:
 def _get_dependency_info() -> dict[str, str]:
     # see the list of dependencies in pyproject.toml
     opt_deps = [
-        "adbc_driver_sqlite",
+        "adbc_driver_manager",
```
Comment (stinodego) [inline py-polars/polars/utils/show_versions.py:62]: Why was this changed from `adbc_driver_sqlite`? Now it doesn't match `pyproject.toml`.

**Review by stinodego — COMMENTED** [review]
Great stuff as always! Some minor remarks/questions.

File: py-polars/polars/utils/show_versions.py (hunk 2 above)
Comment (alexander-beedie) [reply py-polars/polars/utils/show_versions.py:62]: Ahh, this appears to be my (unfinished) PR for ADBC fixes accidentally leaking into this commit - will fix it up 😅

File: py-polars/tests/unit/io/test_spreadsheet.py [hunk 3]
```diff
@@ -223,10 +243,26 @@ def test_schema_overrides_11161(excel_file_path: Path) -> None:
         "iter_groups": pl.Float32,
     }
 
+    for workbook_path in (path_xlsx, path_xlsb, path_ods):
+        print(workbook_path)
```
[the remark this answers is not in the document]
Comment (alexander-beedie) [reply py-polars/tests/unit/io/test_spreadsheet.py]: Yup!

File: py-polars/tests/unit/io/test_spreadsheet.py (hunk 1 above)
Comment (alexander-beedie) [reply py-polars/tests/unit/io/test_spreadsheet.py]: Done ✌️

**Review by stinodego — APPROVED** [review]
All right, good to go!
